### PR TITLE
Rename from nvprof focused connector to nvtx focused connector 

### LINKED ISSUE #176: Fix Nvprof focused connector naming to nvtx, consistent with nvtx connector
Fix Nvprof focused connector naming to nvtx and associated cmakeLists.txt file, and ensure consistency in naming and functionality 

The nvprof-connector is already being renamed to nvtx-connector in #139. This PR is independent of changes to the nvtx connector renaming. 

NOTE: The KOKKOSTOOLS_HAS_NVPROF environment variable should probably be changed to KOKKOSTOOLS_HAS_NVTX as well. I am putting that aside for another PR.

### PR DESCRIPTION
**Pull request by vlkale**
Fix #176.

Fix naming of files and name of tool.

### CODE REVIEW

**Review by dalg24 — CHANGES_REQUESTED** [review]
Your PR description claims the existing tool is being renamed, but the patch is a pure addition alongside the old facility.  Also the build system probably needs some update to accommodate that renaming.

**Comment by vlkale** [comment]
@dalg24 Thanks.This is a draft PR. I ought to have marked it as so. Your comments are on my todo already.

File: profiling/all/kp_all.cpp [hunk 1]
```diff
@@ -87,9 +87,8 @@ EventSet get_event_set(const char* profiler, const char* config_str) {
   handlers["caliper"] = cali::get_kokkos_event_set(config_str);
 #endif
 #ifdef KOKKOSTOOLS_HAS_NVPROF
-  handlers["nvprof-connector"] = NVProfConnector::get_event_set();
-  handlers["nvprof-focused-connector"] =
-      NVProfFocusedConnector::get_event_set();
+  handlers["nvprof-connector"]       = NVProfConnector::get_event_set();
```
Comment (vlkale) [inline profiling/all/kp_all.cpp:94]: This is being fixed in #139

**Comment by vlkale** [comment]
I made the fixes to this PR to get rid of the duplicated files, e.g., nvprof_focused_connector.cpp that are mistakenly sitting alongside the nvtx_* named files. I have also updated the CMakeLists.txt file to read the environment variable KokkosTools_ENABLE_NVTX, which I think would be needed for this PR's renaming, though (as mentioned in one of my comments), other related renaming changes for the CMakeLists.txt - specifically those involving nvprof_connector - are the action items of PR #139.

**Comment by dalg24** [comment]
```
> git rev-parse HEAD
fed7b89985be621b7e07cc27646842fb57a5eb0f

> tree profiling
profiling
[...]
├── nvprof-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvprof_connector.cpp
│   └── kp_nvprof_connector_domain.h
├── nvprof-focused-connector                  < This PR claims it is renaming files
│   ├── CMakeLists.txt.                       < but as you can see all of these are still
│   ├── Makefile                              < there.  Please fix it.
│   ├── kp_nvprof_focused_connector.cpp       <
│   └── kp_nvprof_focused_connector_domain.h  <
├── nvtx-focused-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvtx_focused_connector.cpp
│   └── kp_nvtx_focused_connector_domain.h
[...]
23 directories, 88 files

File: profiling/nvtx-focused-connector/kp_nvtx_focused_connector_domain.h [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+#ifndef _H_KOKKOSP_KERNEL_NVPROF_CONNECTOR_INFO
+#define _H_KOKKOSP_KERNEL_NVPROF_CONNECTOR_INFO
```
Comment (dalg24) [inline profiling/nvtx-focused-connector/kp_nvtx_focused_connector_domain.h]:
```suggestion
#ifndef KOKKOSP_NVTX_FOCUSED_CONNECTOR_H
#define KOKKOSP_NVTX_FOCUSED_CONNECTOR_H
```

Comment (vlkale) [reply profiling/nvtx-focused-connector/kp_nvtx_focused_connector_domain.h]: Your suggestion looks good to me at first sight, and I assume this matches naming conventions for `#define`'ed  in other parts of Kokkos Tools/Core/Kernels. 
One question is what the `INFO` at the end of`_H_KOKKOSP_KERNEL_NVPROF_CONNECTOR_INFO` is (which is not in the new environment variable you suggest). I would like to look this over a bit more and Iwill commit soon once I convince myself.

**Comment by vlkale** [comment]
> ```
> > git rev-parse HEAD
> fed7b89985be621b7e07cc27646842fb57a5eb0f
> 
> > tree profiling
> profiling
> [...]
> ├── nvprof-connector
> │   ├── CMakeLists.txt
> │   ├── Makefile
> │   ├── kp_nvprof_connector.cpp
> │   └── kp_nvprof_connector_domain.h
> ├── nvprof-focused-connector                  < This PR claims it is renaming files
> │   ├── CMakeLists.txt.                       < but as you can see all of these are still
> │   ├── Makefile                              < there.  Please fix it.
> │   ├── kp_nvprof_focused_connector.cpp       <
> │   └── kp_nvprof_focused_connector_domain.h  <
> ├── nvtx-focused-connector
> │   ├── CMakeLists.txt
> │   ├── Makefile
> │   ├── kp_nvtx_focused_connector.cpp
> │   └── kp_nvtx_focused_connector_domain.h
> [...]
> 23 directories, 88 files
> ```

@dalg24  OK, got it and thanks for that. 

This ought to be fixed now with latest changes pushed  to the repo - as seen in the below print. However, now there seems to be a merge conflict for `profiling/nvprof-focused-connector/kp_nvprof_focused_connector.cpp`  and 
`profiling/nvprof-focused-connector/kp_nvprof_focused_connector_domain.h`. 


```
user@system:kokkos-tools-develop vlkale$ git rev-parse HEAD
fed7b89985be621b7e07cc27646842fb57a5eb0f
user@system :kokkos-tools-develop vlkale$ tree profiling/
profiling/
├── all
│   ├── CMakeLists.txt
│   ├── impl
│   │   ├── Kokkos_Profiling_C_Interface.h
│   │   ├── Kokkos_Profiling_DeviceInfo.hpp
│   │   └── Kokkos_Profiling_Interface.hpp
│   ├── kp_all.cpp
│   ├── kp_all.hpp
│   └── kp_core.hpp
[...]
├── nvprof-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvprof_connector.cpp
│   └── kp_nvprof_connector_domain.h
├── nvtx-focused-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvtx_focused_connector.cpp
│   └── kp_nvtx_focused_connector_domain.h
[...]
26 directories, 83 files
s1017105ca:kokkos-tools-develop vlkale$ 

```

File: profiling/nvtx-focused-connector/kp_nvtx_focused_connector_domain.h (hunk 2 above)
Comment (vlkale) [reply profiling/nvtx-focused-connector/kp_nvtx_focused_connector_domain.h]: Fixed

**Comment by vlkale** [comment]
> I made the fixes to this PR to get rid of the duplicated files, e.g., nvprof_focused_connector.cpp that are mistakenly sitting alongside the nvtx_* named files. I have also updated the CMakeLists.txt file to read the environment variable KokkosTools_ENABLE_NVTX, which I think would be needed for this PR's renaming, though (as mentioned in one of my comments), other related renaming changes for the CMakeLists.txt - specifically those involving nvprof_connector - are the action items of PR #139.

After thinking again, I have decided to actually put this is another PR (possibly PR #139) as it is not clear if it may interfere with other tools.

**Comment by vlkale** [comment]
FYI: I have resolved the merge conflicts a few days ago and this is building properly now against the CI tests. Please let me know if there is anything else. I think this is ready unless there are any other comments here. See the result of tree below. 

@dalg24  

```
user@system kto-dev-vk % git rev-parse HEAD                                         
50c1b1b0481c75b829f27631a21375c6bbf7eeac
user@system kto-dev-vk % tree profiling                                             
profiling
├── all
│   ├── CMakeLists.txt
│   ├── impl
│   │   ├── Kokkos_Profiling_C_Interface.h
│   │   ├── Kokkos_Profiling_DeviceInfo.hpp
│   │   └── Kokkos_Profiling_Interface.hpp
│   ├── kp_all.cpp
│   ├── kp_all.hpp
│   └── kp_core.hpp
├── caliper-connector
│   └── README.md
├── chrome-tracing
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_chrome_tracing.cpp
├── memory-events
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_memory_events.cpp
│   ├── kp_memory_events.hpp
│   └── kp_timer.hpp
├── memory-hwm
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_hwm.cpp
├── memory-hwm-mpi
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_hwm_mpi.cpp
├── memory-usage
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_memory_usage.cpp
├── nvprof-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvprof_connector.cpp
│   └── kp_nvprof_connector_domain.h
├── nvtx-focused-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_nvtx_focused_connector.cpp
│   └── kp_nvtx_focused_connector_domain.h
├── papi-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── README.md
│   ├── kp_papi_connector.cpp
│   └── kp_papi_connector_domain.h
├── perfetto-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── README.md
│   ├── kokkos.cfg
│   ├── libperfetto-connector.cpp
│   └── perfetto
│       ├── LICENSE
│       ├── perfetto.cc
│       └── perfetto.h
├── roctx-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_roctx_connector.cpp
├── simple-kernel-timer
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_json_writer.cpp
│   ├── kp_kernel_info.h
│   ├── kp_kernel_timer.cpp
│   ├── kp_kernel_timer_json.cpp
│   ├── kp_reader.cpp
│   ├── kp_shared.cpp
│   └── kp_shared.h
├── space-time-stack
│   ├── CMakeLists.txt
│   ├── Makefile
│   └── kp_space_time_stack.cpp
├── systemtap-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kokkos_stap_deep_copy.stp
│   ├── kokkos_stap_example1.stp
│   ├── kokkos_stap_mon.stp
│   ├── kp_systemtap_connector.cpp
│   └── probes.d
├── timemory-connector
│   └── README.md
├── variorum-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── README.md
│   └── variorum-connector.cpp
├── vtune-connector
│   ├── CMakeLists.txt
│   ├── Makefile
│   ├── kp_vtune_connector.cpp
│   └── kp_vtune_connector_domain.h
└── vtune-focused-connector
    ├── CMakeLists.txt
    ├── Makefile
    ├── kp_vtune_focused_connector.cpp
    └── kp_vtune_focused_connector_domain.h

22 directories, 81 files

```

**Pull request closed without merging**